src/main/java/com/google/sps/TextDetectionAPIImpl.java

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.cloud.vision.v1.AnnotateImageRequest;
+import com.google.cloud.vision.v1.AnnotateImageResponse;
+import com.google.cloud.vision.v1.BatchAnnotateImagesResponse;
+import com.google.cloud.vision.v1.EntityAnnotation;
+import com.google.cloud.vision.v1.Image;
+import com.google.cloud.vision.v1.ImageAnnotatorClient;
+import com.google.sps.data.ShoppingListTextEntry;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TextDetectionAPIImpl implements TextDetectionAPI {
+  public List<ShoppingListTextEntry> detect(byte[] imageBytes) throws PhotoDetectionException {
+    Image shoppingImage = PhotoShoppingUtil.getImageFromBytes(imageBytes);
+
+    List<AnnotateImageRequest> requests = generateShoppingImageRequest(shoppingImage);
+
+    BatchAnnotateImagesResponse response = detectTextFromImage(requests);
+
+    return parseAnnotateImageResponse(response);
+  }
+
+  /** Generates the request query to be sent to CloudVisionAPI client. */
+  private List<AnnotateImageRequest> generateShoppingImageRequest(Image shoppingImage) {
+    List<AnnotateImageRequest> requests = new ArrayList<>();
```

Comment:
The return value does not need to be a list since you always return one item.

Reply:
Totally agree that a list is not required.
Please correct me if I'm wrong, I checked the [documentation](https://googleapis.dev/nodejs/vision/latest/v1p1beta1.ImageAnnotatorClient.html) and it says that the client accepts only a list of request as a parameter. And the client doesn't have any other method from which we can get a response. At least not the `ImageAnnotatorClient`.

Reply:
You are right. There's no documentation for a single image annotation.